EXP-1865 add isRollout boolean property to Experiment type

lmorchard opened the pull request:
Because:

* we want to implement support for an isRollout flag in experiments

This commit:

* adds an isRollout boolean to the NimbusExperiment type

* introduces example recipes for both mobile and desktop for testing

lmorchard commented:
Kind of winging it on the example files as my first try. Let's see what I missed!

lmorchard commented on CHANGELOG.md:
```diff
@@ -4,7 +4,7 @@
 
 ### Minor Changes
 
-- Updating package.json 
```
Hmm. Not something I changed in this PR, but it's failing CI

jaredlockhart commented on data/experiment-recipe-samples/rollout-desktop.json:
```diff
@@ -0,0 +1,42 @@
+{
```
We should include a desktop version number in this filename, I think it's 95 but maybe confirm with @piatra

piatra replied:
95 if this lands before next Tuesday Dec 7th but I think less important because Experimenter can make changes outside of the train model. What people are interested in is when both Experimenter and Firefox support X feature so the answer most likely will come from the Firefox side.

piatra replied:
In this case we'll land the Firefox code in 97 so in practice we won't be able to launch/test except on >= 97.

jaredlockhart commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+{
```
Also figure out a minimum supported version from mobile which is tricky becuase it varies by application (ie fenix/ios/focus/klar/etc)

piatra commented on data/experiment-recipe-samples/rollout-desktop.json:
```diff
@@ -0,0 +1,42 @@
+  "userFacingName": "Testing Rollout Desktop",
+  "referenceBranch": "control",
+  "proposedDuration": 28,
+  "isRollout": true,
```
oh, I have this implemented as `is_rollout` in my current Desktop patch (not yet landed). This is what we're going with `isRollout` right?

lmorchard replied:
I'd say yeah, this is what we're going with - if only to follow the camelCase convention of the other properties

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+  "slug": "rollout-mobile-example",
+  "application": "reference_browser",
+  "appId": "org.mozilla.reference.browser",
+  "appName": "reference_browser",
```
Should this be more specific? If we test both iOS and Android forms, we would need to have both Firefox versions.

jaredlockhart replied:
Good point @heres-maria maybe we should create a sample recipe for each of desktop/ios/android.

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+  "application": "reference_browser",
+  "appId": "org.mozilla.reference.browser",
+  "appName": "reference_browser",
+  "featureIds": ["bookmark-icon"],
```
I have not seen this feature in Stage.

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+    "randomizationUnit": "nimbus_id",
+    "namespace": "rollout-mobile-example-rollout-1",
+    "start": 0,
+    "count": 5000,
```
We would normally test with a rollout created for 100% of the population, as it is the most likely scenario.

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+  "isEnrollmentPaused": false,
+  "bucketConfig": {
+    "randomizationUnit": "nimbus_id",
+    "namespace": "rollout-mobile-example-rollout-1",
```
Is this the correct namespace?

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+    "count": 5000,
+    "total": 10000
+  },
+  "targeting": null,
```
Should targeting be "true" instead?

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+      "slug": "control",
+      "ratio": 1,
+      "feature": {
+        "featureId": "bookmarkId",
```
Similar to my comment above, I have not seen this feature in Stage.

lmorchard replied:
Do you have sample data that could be used here instead? This is just a copy of [another data file in this repo](https://github.com/mozilla/nimbus-shared/blob/main/data/experiment-recipe-samples/mobile-a-a.json) with `isRollout: true` and a few other minor changes.

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+  "id": "rollout-mobile-example",
+  "slug": "rollout-mobile-example",
+  "application": "reference_browser",
+  "appId": "org.mozilla.reference.browser",
```
I was expecting this to be specific, but from what I understand from @jhugman, the value of appId is redundant

heres-maria commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+  "schemaVersion": "0.1.0",
+  "id": "rollout-mobile-example",
+  "slug": "rollout-mobile-example",
+  "application": "reference_browser",
```
I was expecting this to be specific, but from what I understand from @jhugman, the value of the application is redundant

lmorchard commented on data/experiment-recipe-samples/rollout-desktop.json:
```diff
@@ -0,0 +1,42 @@
+    "namespace": "firefox-desktop-no-feature-firefox-desktop-rollout-1",
+    "randomizationUnit": "normandy_id"
+  },
+  "schemaVersion": "1.5.0",
```
Removing this per #186

lmorchard commented on data/experiment-recipe-samples/rollout-mobile.json:
```diff
@@ -0,0 +1,38 @@
+{
+  "schemaVersion": "0.1.0",
```
Removing this per #186

lmorchard commented:
Seeing all the comments on the example data in this PR, it seems maybe worth backing up and seeing if we have a better source for these?

Maybe @heres-maria has a better way to come up with this sample data?

My goal was just to set `isRollout: true` with the rest of the experiment being unremarkable. Most of the rest of the properties are copied over from [other data files in this repo](https://github.com/mozilla/nimbus-shared/tree/main/data/experiment-recipe-samples), but it sounds like those are problematic also?

heres-maria commented:
> Seeing all the comments on the example data in this PR, it seems maybe worth backing up and seeing if we have a better source for these?
> 
> Maybe @heres-maria has a better way to come up with this sample data? That is - come up what is actually useful for testing these things, put that in this PR.
> 
> My goal was just to set `isRollout: true` with the rest of the experiment being unremarkable. Most of the rest of the properties are copied over from [other data files in this repo](https://github.com/mozilla/nimbus-shared/tree/main/data/experiment-recipe-samples), but it sounds like those are problematic also?

@lmorchard For the feature part at least, could these production experiments be used as the starting point for the iOS and Fenix recipes? They are current rollouts. 
https://experimenter.services.mozilla.com/nimbus/all-users-homescreen-ios-production
https://experimenter.services.mozilla.com/nimbus/all-users-home-screen-android-production

lmorchard commented on data/experiment-recipe-samples/desktop-97-rollout.json:
```diff
@@ -0,0 +1,48 @@
+  "appName": "firefox_desktop",
+  "application": "firefox-desktop",
+  "arguments": {},
+  "isRollout": true,
```
Added this new property, which is the main event in this PR.

lmorchard commented on data/experiment-recipe-samples/desktop-97-rollout.json:
```diff
@@ -0,0 +1,48 @@
+  ],
+  "bucketConfig": {
+    "count": 1700,
+    "namespace": "urlbar-21-rollout",
```
Added `-rollout` to the namespace, to simulate [the nimbus model logic here](https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/models/nimbus.py#L422-L423)

lmorchard commented on data/experiment-recipe-samples/desktop-97-rollout.json:
```diff
@@ -0,0 +1,48 @@
+  "proposedEnrollment": 7,
+  "referenceBranch": "control",
+  "schemaVersion": "1.7.0",
+  "slug": "desktop-rollout-example-firefox-suggest-history-rollout-3",
```
Tweaked the slug and IDs

lmorchard commented on data/experiment-recipe-samples/desktop-97-rollout.json:
```diff
@@ -0,0 +1,48 @@
+  "proposedDuration": 28,
+  "proposedEnrollment": 7,
+  "referenceBranch": "control",
+  "schemaVersion": "1.7.0",
```
No idea what the eventual schemaVersion should be, here. It depends on what PR lands first, and we'd like to remove it anyway per issue #186 - but we need *something* here to pass CI in the meantime

lmorchard reviewed (commented):
Alright, another stab at maybe some useful test files? Not sure if these are it.

Main thing is that I copied the recipes for these experiments and tried hand-tweaking them to add rollout properties as noted in comments:

* https://experimenter.services.mozilla.com/nimbus/firefox-suggest-history-rollout-3/details
* https://experimenter.services.mozilla.com/nimbus/all-users-homescreen-ios-production/details
* https://experimenter.services.mozilla.com/nimbus/all-users-home-screen-android-production/details

piatra reviewed (approved):
Desktop recipe looks good. Thanks.

lmorchard commented:
Alright, maybe third try's the charm? Rebased off main, copied the multifeature example data files, changed some names and added `"isRollout": true` to them along with adjusting the bucket namespace. Not sure what else needs doing?

heres-maria commented on data/experiment-recipe-samples/ios-39-rollout.json:
```diff
@@ -0,0 +1,85 @@
+  "appName": "firefox_ios",
+  "application": "org.mozilla.ios.Firefox",
+  "arguments": {},
+  "branches": [
```
This has multiple branches and affects multiple features. Should this be single branch and single feature?

heres-maria replied:
We would also like to test the interaction between rollouts and a multifeature experiment of this type, but that would have isRollout false.

lmorchard replied:
@heres-maria Can you compose the test data that would actually work for testing? We could add it to the repo after testing. I've tried a few times now in this PR - none of those tries has passed muster. Would rather just record what was used to successfully test the feature than keep guessing at what might work

lmorchard commented:
Revised this PR to drop the sample data. The schema change here is very small, but the testing data blocking merge of the PR seems to have been a moving target that I don't have the context to assemble properly.

Maybe we should just add the sample data after successful testing? That way we record what was used rather than trying to guess what might work?

lmorchard commented:
Missed the thumbs up earlier, just going to merge this and we can move on with sample data when we have it

heres-maria commented:
Possible testing data:
**Desktop**:
<details>
  <summary>Desktop Rollout with pocketNewtab enabled</summary>

```
{
  "slug": "firefox-desktop-rollout-test-final",
  "appId": "firefox-desktop",
  "appName": "firefox_desktop",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "feature": {
        "value": {},
        "enabled": true,
        "featureId": "unused-feature-id-for-legacy-support"
      },
      "features": [
        {
          "value": {
            "loadMore": true,
            "compactLayout": true,
            "newFooterSection": true,
            "lastCardMessageEnabled": true
          },
          "enabled": true,
          "featureId": "pocketNewtab"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": true,
  "probeSets": [],
  "startDate": "2021-10-26",
  "targeting": "true",
  "featureIds": [
    "pocketNewtab"
  ],
  "application": "firefox-desktop",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "firefox-desktop-pocketNewtab-release-rollout",
    "randomizationUnit": "normandy_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Firefox Desktop Rollout Test Pocket",
  "referenceBranch": "control",
  "proposedDuration": 63,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Rollout for pocketNewTab",
  "id": "firefox-desktop-rollout-test-final",
  "last_modified": 1639560912253
}
```

</details>
<details>
  <summary>Desktop Rollout with privatebrowsing disabled</summary>

```
{
  "slug": "firefox-desktop-rollout-test-final-2",
  "appId": "firefox-desktop",
  "appName": "firefox_desktop",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "feature": {
        "value": {},
        "enabled": false,
        "featureId": "unused-feature-id-for-legacy-support"
      },
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "privatebrowsing"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": true,
  "probeSets": [],
  "startDate": "2021-10-26",
  "targeting": "true",
  "featureIds": [
    "privatebrowsing"
  ],
  "application": "firefox-desktop",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "firefox-desktop-privatebrowsing-release-rollout",
    "randomizationUnit": "normandy_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Firefox Desktop Rollout Test P",
  "referenceBranch": "control",
  "proposedDuration": 63,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Rollout for privatebrowsing",
  "id": "firefox-desktop-rollout-test-final-2",
  "last_modified": 1639560912253
}
```

</details>
<details>
  <summary>Desktop Experiment with isRollout false pocketNewtab enabled</summary>

```
{
  "slug": "firefox-desktop-rollout-experiment1",
  "appId": "firefox-desktop",
  "appName": "firefox_desktop",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "feature": {
        "value": {},
        "enabled": true,
        "featureId": "unused-feature-id-for-legacy-support"
      },
      "features": [
        {
          "value": {
            "loadMore": true,
            "compactLayout": true,
            "newFooterSection": true,
            "lastCardMessageEnabled": true
          },
          "enabled": true,
          "featureId": "pocketNewtab"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": "2021-10-26",
  "targeting": "true",
  "featureIds": [
    "pocketNewtab"
  ],
  "application": "firefox-desktop",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "firefox-desktop-pocketNewtab-release-1",
    "randomizationUnit": "normandy_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Firefox Desktop Experiment 1 Pocket enabled",
  "referenceBranch": "control",
  "proposedDuration": 63,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment 1 enabled for pocketNewTab",
  "id": "firefox-desktop-rollout-experiment1",
  "last_modified": 1639560912253
}
```

</details>
<details>
  <summary>Desktop Experiment with isRollout false pocketNewtab disabled</summary>

```
{
  "slug": "firefox-desktop-rollout-experiment2",
  "appId": "firefox-desktop",
  "appName": "firefox_desktop",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "feature": {
        "value": {},
        "enabled": false,
        "featureId": "unused-feature-id-for-legacy-support"
      },
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "pocketNewtab"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": "2021-10-26",
  "targeting": "true",
  "featureIds": [
    "pocketNewtab"
  ],
  "application": "firefox-desktop",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "firefox-desktop-pocketNewtab-release-2",
    "randomizationUnit": "normandy_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Firefox Desktop Experiment 2 Pocket disabled",
  "referenceBranch": "control",
  "proposedDuration": 63,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment 2 disabled for pocketNewTab",
  "id": "firefox-desktop-rollout-experiment2",
  "last_modified": 1639560912253
}
```

</details>
<details>
  <summary>Desktop Multifeature Experiment with isRollout false pocketNewtab and privatebrowsing enabled</summary>

```
{
  "slug": "firefox-desktop-rollout-experimentm1",
  "appId": "firefox-desktop",
  "appName": "firefox_desktop",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "feature": {
        "value": {},
        "enabled": false,
        "featureId": "unused-feature-id-for-legacy-support"
      },
      "features": [
        {
          "value": {
            "loadMore": true,
            "compactLayout": true,
            "newFooterSection": true,
            "lastCardMessageEnabled": true
          },
          "enabled": true,
          "featureId": "pocketNewtab"
        },
        {
          "value": {
            "infoEnabled": true,
            "infoTitle": "Hello world",
            "infoBody": "This is some text",
            "infoLinkText": "This is a link",
            "infoLinkUrl": "http://foo.mozilla.com/%LOCALE%",
            "infoIcon": "chrome://branding/content/about-logo.png",
            "promoEnabled": true,
            "promoTitle": "Promo title",
            "promLinkText": "Promo link",
            "promoLinkUrl": "https://vpn.mozilla.org/?utm_source=firefox-browser&utm_medium=firefox-%CHANNEL%-browser&utm_campaign=private-browsing-vpn-link&utm_experiment=private-browsing-test&utm_branch=treatment"
          },
          "enabled": true,
          "featureId": "privatebrowsing"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": "2021-10-26",
  "targeting": "true",
  "featureIds": [
    "pocketNewtab",
    "privatebrowsing"
  ],
  "application": "firefox-desktop",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "firefox-desktop-pocketNewtab-privatebrowsing-release-1",
    "randomizationUnit": "normandy_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Experiment 1 multifeature enabled",
  "referenceBranch": "control",
  "proposedDuration": 63,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment 1 multifeature enabled",
  "id": "firefox-desktop-rollout-experimentm1",
  "last_modified": 1639560912253
}
```

</details>
<details>
  <summary>Desktop Multifeature Experiment with isRollout false pocketNewtab and privatebrowsing disabled</summary>

```
{
  "slug": "firefox-desktop-rollout-experimentm2",
  "appId": "firefox-desktop",
  "appName": "firefox_desktop",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "feature": {
        "value": {},
        "enabled": false,
        "featureId": "unused-feature-id-for-legacy-support"
      },
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "pocketNewtab"
        },
        {
          "value": {},
          "enabled": false,
          "featureId": "privatebrowsing"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": "2021-10-26",
  "targeting": "true",
  "featureIds": [
    "pocketNewtab",
    "privatebrowsing"
  ],
  "application": "firefox-desktop",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "firefox-desktop-pocketNewtab-privatebrowsing-release-2",
    "randomizationUnit": "normandy_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Experiment 2 multifeature disabled",
  "referenceBranch": "control",
  "proposedDuration": 63,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment 2 multifeature disabled",
  "id": "firefox-desktop-rollout-experimentm2",
  "last_modified": 1639560912253
}
```

</details>

**Firefox Android**:
<details>
  <summary>Android Rollout with homescreen enabled</summary>

```
{
  "slug": "firefox-android-rollout-test-final",
  "appId": "org.mozilla.firefox",
  "appName": "fenix",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {
            "sections-enabled": {
              "pocket": false,
              "topSites": false,
              "jumpBackIn": true,
              "recentlySaved": false,
              "recentExplorations": false
            }
          },
          "enabled": true,
          "featureId": "homescreen"
        }
      ]
    }
  ],
  "outcomes": [
    {
      "slug": "default-browser",
      "priority": "primary"
    }
  ],
  "arguments": {},
  "isRollout": true,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen"
  ],
  "application": "org.mozilla.firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "fenix-homescreen-release-rollout",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Homepage Android Rollout Test 1",
  "referenceBranch": "control",
  "proposedDuration": 45,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 28,
  "userFacingDescription": "Android ROllout Homescreen",
  "id": "firefox-android-rollout-test-final",
  "last_modified": 1639991916998
}
```

</details>
<details>
  <summary>Android Rollout with homescreen disabled</summary>

```
{
  "slug": "firefox-android-rollout-test-final-2",
  "appId": "org.mozilla.firefox",
  "appName": "fenix",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "homescreen"
        }
      ]
    }
  ],
  "outcomes": [
    {
      "slug": "default-browser",
      "priority": "primary"
    }
  ],
  "arguments": {},
  "isRollout": true,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen"
  ],
  "application": "org.mozilla.firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "fenix-homescreen-release-rollout",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Homepage Android Rollout Test 2",
  "referenceBranch": "control",
  "proposedDuration": 45,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 28,
  "userFacingDescription": "Android Rollout Homescreen disabled",
  "id": "firefox-android-rollout-test-final-2",
  "last_modified": 1639991916998
}
```

</details>
<details>
  <summary>Android Experiment with isRollout false homescreen enabled</summary>

```
{
  "slug": "firefox-android-homepage-experiment-1",
  "appId": "org.mozilla.firefox",
  "appName": "fenix",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {
            "sections-enabled": {
              "pocket": true,
              "topSites": false,
              "jumpBackIn": true,
              "recentlySaved": false,
              "recentExplorations": false
            }
          },
          "enabled": true,
          "featureId": "homescreen"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen"
  ],
  "application": "org.mozilla.firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "fenix-homescreen-release-3",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Homepage Android Experiment 1",
  "referenceBranch": "control",
  "proposedDuration": 45,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 28,
  "userFacingDescription": "Android Homepage on with a single difference",
  "id": "firefox-android-homepage-experiment-1",
  "last_modified": 1639991916998
}
```

</details>
<details>
  <summary>Android Experiment with isRollout false homescreen disabled</summary>

```
{
  "slug": "firefox-android-homepage-experiment-2",
  "appId": "org.mozilla.firefox",
  "appName": "fenix",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "homescreen"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen"
  ],
  "application": "org.mozilla.firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "fenix-homescreen-release-4",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Homepage Android Experiment 2",
  "referenceBranch": "control",
  "proposedDuration": 45,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 28,
  "userFacingDescription": "Android Homepage off",
  "id": "firefox-android-homepage-experiment-2",
  "last_modified": 1639991916998
}
```

</details>
<details>
  <summary>Android Multifeature Experiment with isRollout false homescreen and fenix-default-browser enabled</summary>

```
{
  "slug": "firefox-android-homepage-experiment-m1",
  "appId": "org.mozilla.firefox",
  "appName": "fenix",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "default_browser_settings_menu",
      "ratio": 1,
      "features": [
        {
          "value": {
            "sections-enabled": {
              "pocket": true,
              "topSites": false,
              "jumpBackIn": true,
              "recentlySaved": false,
              "recentExplorations": false
            }
          },
          "enabled": true,
          "featureId": "homescreen"
        },
        {
          "value": {},
          "enabled": true,
          "featureId": "fenix-default-browser"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "fenix-default-browser",
    "homescreen"
  ],
  "application": "org.mozilla.firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "fenix-fenix-default-browser-homescreen-release-3",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Homepage Android Experiment Multifeature 1",
  "referenceBranch": "default_browser_settings_menu",
  "proposedDuration": 45,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 28,
  "userFacingDescription": "Android Homepage on with a single difference multifeature",
  "id": "firefox-android-homepage-experiment-m1",
  "last_modified": 1639991916998
}
```

</details>
<details>
  <summary>Android Multifeature Experiment with isRollout false homescreen and fenix-default-browser disabled</summary>

```
{
  "slug": "firefox-android-homepage-experiment-m2",
  "appId": "org.mozilla.firefox",
  "appName": "fenix",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "default_browser_settings_menu",
      "ratio": 1,
      "features": [
        {
          "value": {
            "sections-enabled": {
              "pocket": true,
              "topSites": false,
              "jumpBackIn": true,
              "recentlySaved": false,
              "recentExplorations": false
            }
          },
          "enabled": true,
          "featureId": "homescreen"
        },
        {
          "value": {},
          "enabled": true,
          "featureId": "fenix-default-browser"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "fenix-default-browser",
    "homescreen"
  ],
  "application": "org.mozilla.firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "fenix-fenix-default-browser-homescreen-release-3",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "Homepage Android Experiment Multifeature 2",
  "referenceBranch": "default_browser_settings_menu",
  "proposedDuration": 45,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 28,
  "userFacingDescription": "Android Homepage on with a single difference multifeature 2",
  "id": "firefox-android-homepage-experiment-m2",
  "last_modified": 1639991916998
}
```

</details>

**Firefox iOS**:
<details>
  <summary>iOS Rollout with homescreen enabled</summary>

```
{
  "slug": "firefox-ios-rollout-test-final",
  "appId": "org.mozilla.ios.Firefox",
  "appName": "firefox_ios",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {
            "jumpBackIn": false,
            "libraryShortcuts": true,
            "recentlySaved": true
          },
          "enabled": true,
          "featureId": "homescreen"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": true,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen"
  ],
  "application": "org.mozilla.ios.Firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "ios-homescreen-release-rollout",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "iOS Rollout 1",
  "referenceBranch": "control",
  "proposedDuration": 28,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Rollout for homescreen enabled feature",
  "id": "firefox-ios-rollout-test-final",
  "last_modified": 1639561341272
}
```

</details>
<details>
  <summary>iOS Rollout with homescreen disabled</summary>

```
{
  "slug": "firefox-ios-rollout-test-final-2",
  "appId": "org.mozilla.ios.Firefox",
  "appName": "firefox_ios",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "onboarding-default-browser"
        }
      ]
    }
  ],
  "outcomes": [
    {
      "slug": "default_browser",
      "priority": "primary"
    }
  ],
  "arguments": {},
  "isRollout": true,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "onboarding-default-browser"
  ],
  "application": "org.mozilla.ios.Firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "ios-onboarding-default-browser-release-rollout",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "iOS Rollout 2",
  "referenceBranch": "control",
  "proposedDuration": 28,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Rollout for onboarding-default-browser disabled feature",
  "id": "firefox-ios-rollout-test-final-2",
  "last_modified": 1639561341272
}
```
</details>
<details>
<summary>iOS Experiment with isRollout false homescreen enabled</summary>

```
{
  "slug": "firefox-ios-homepage-experiment-1",
  "appId": "org.mozilla.ios.Firefox",
  "appName": "firefox_ios",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {
            "jumpBackIn": false,
            "libraryShortcuts": true,
            "recentlySaved": true
          },
          "enabled": true,
          "featureId": "homescreen"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen"
  ],
  "application": "org.mozilla.ios.Firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "ios-homescreen-release-3",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "iOS Single Feature Experiment 1,
  "referenceBranch": "control",
  "proposedDuration": 28,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment for homescreen enabled feature",
  "id": "firefox-ios-homepage-experiment-1",
  "last_modified": 1639561341272
}
```

</details>
<details>
  <summary>iOS Experiment with isRollout false homescreen disabled</summary>

```{
  "slug": "firefox-ios-homepage-experiment-2",
  "appId": "org.mozilla.ios.Firefox",
  "appName": "firefox_ios",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "homescreen"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen"
  ],
  "application": "org.mozilla.ios.Firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "ios-homescreen-release-4",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "iOS Single Feature Experiment 2,
  "referenceBranch": "control",
  "proposedDuration": 28,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment for homescreen disabled feature",
  "id": "firefox-ios-homepage-experiment-2",
  "last_modified": 1639561341272
}
```

</details>
<details>
  <summary>iOS Multifeature Experiment with isRollout false homescreen and fenix-default-browser enabled</summary>

```
{
  "slug": "firefox-ios-homepage-experiment-m1",
  "appId": "org.mozilla.ios.Firefox",
  "appName": "firefox_ios",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "default_browser_newtab_banner",
      "ratio": 1,
      "features": [
        {
          "value": {
            "jumpBackIn": false,
            "libraryShortcuts": true,
            "recentlySaved": true
          },
          "enabled": true,
          "featureId": "homescreen"
        },
        {
          "value": {},
          "enabled": true,
          "featureId": "onboarding-default-browser"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen",
    "onboarding-default-browser"
  ],
  "application": "org.mozilla.ios.Firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "ios-homescreen-onboarding-default-browser-release-3",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "iOS Single Feature Experiment Multifeature 1",
  "referenceBranch": "control",
  "proposedDuration": 28,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment multifeature for homescreen + onboarding-default-browser enabled features",
  "id": "firefox-ios-homepage-experiment-m1",
  "last_modified": 1639561341272
}
```

</details>
<details>
  <summary>iOS Multifeature Experiment with isRollout false homescreen and fenix-default-browser disabled</summary>

```
{
  "slug": "firefox-ios-homepage-experiment-m2",
  "appId": "org.mozilla.ios.Firefox",
  "appName": "firefox_ios",
  "channel": "release",
  "endDate": null,
  "branches": [
    {
      "slug": "control",
      "ratio": 1,
      "features": [
        {
          "value": {},
          "enabled": false,
          "featureId": "homescreen"
        },
        {
          "value": {},
          "enabled": false,
          "featureId": "onboarding-default-browser"
        }
      ]
    }
  ],
  "outcomes": [],
  "arguments": {},
  "isRollout": false,
  "probeSets": [],
  "startDate": null,
  "targeting": "true",
  "featureIds": [
    "homescreen",
    "onboarding-default-browser"
  ],
  "application": "org.mozilla.ios.Firefox",
  "bucketConfig": {
    "count": 10000,
    "start": 0,
    "total": 10000,
    "namespace": "ios-homescreen-onboarding-default-browser-release-4",
    "randomizationUnit": "nimbus_id"
  },
  "schemaVersion": "1.6.2",
  "userFacingName": "iOS Single Feature Experiment Multifeature 2",
  "referenceBranch": "control",
  "proposedDuration": 28,
  "isEnrollmentPaused": false,
  "proposedEnrollment": 7,
  "userFacingDescription": "Experiment multifeature for homescreen + onboarding-default-browser disabled features",
  "id": "firefox-ios-homepage-experiment-m2",
  "last_modified": 1639561341272
}
```

</details>
